### PR TITLE
codecoverage/bot: Add tests for artifacts.generate_path

### LINKED ISSUE #911: Write tests for the artifacts module
At the moment, only a subset of the functions of the module are tested.

- [ ] `generate_path`;
- [x] `get_chunks`;
- [x] `get`;
- [x] `download`;
- [x] `download_all`.

### PR DESCRIPTION
**Pull request by strodew**
Uses dict to mock JSON file containing artifact names that would ordinarily be available to the function when it is called in artifacts.py (potentially a better way would be to add a fixture for one of the testing JSON files used for testing downloadall and also use that for generate_path but I was not sure if that would be wise)

At any rate, pytest tests/test_artifacts.py runs with all passes on my end. I am unable to run ./please check due to lack of taskcluster credentials so I wasn't sure how to get that run but my changes shouldn't interfere with anything else. 

I diff'd the repo and it seemed to think I'd moved the FAKE_ARTIFACTS_DIR fixture code but I wasn't able to identify how if I had (pretty sure I didn't) so hopefully that's not an issue.

Fixes #911.

### CODE REVIEW

File: src/codecoverage/bot/tests/test_artifacts.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ def FAKE_ARTIFACTS_DIR(tmpdir):
     return tmpdir.strpath
 
 
+def test_generate_path(FAKE_ARTIFACTS_DIR):
+    a = ArtifactsHandler([], parent_dir=FAKE_ARTIFACTS_DIR)
+    artifact = {'name': 'code-coverage-jsvm.info'}
```
Comment (marco-c) [inline src/codecoverage/bot/tests/test_artifacts.py]: Nit: assign a value to `artifact` then test generate_path, then overwrite `artifact` and test it again. This way we don't need the `artifact2` variable. The other option is to rename `artifact` to `artifact_jsvm` and `artifact2` to `artifact_grcov`, which would make the variable names more meaningful.

File: src/codecoverage/bot/tests/test_artifacts.py [hunk 2]
```diff
@@ -27,6 +27,14 @@ def FAKE_ARTIFACTS_DIR(tmpdir):
...
+    artifact2 = {'name': 'code-coverage-grcov.zip'}
+    assert a.parent_dir + '/linux_xpcshell-3_code-coverage-jsvm.info' == a.generate_path('linux', 'xpcshell-3', artifact)
+    assert a.parent_dir + '/windows_cppunit_code-coverage-grcov.zip' == a.generate_path('windows', 'cppunit', artifact2)
```
Comment (marco-c) [inline src/codecoverage/bot/tests/test_artifacts.py]: Nit: using os.path.join would be preferable, instead of manually building the path

**Comment by strodew** [comment]
Hmm, fixed the nits but I think I might have locked my ability to push the changes by clicking on the update branch button? Not sure. I can always add it as a further pull request later but if there's a way to neaten that up then that probably makes sense.

**Comment by marco-c** [comment]
By updating the branch, you've merged master into your remote branch. Now your local branch and your remote branch have diverged, so git is rejecting your push.
You can force the push to overwrite the remote branch, `git push origin generate_path_test --force`.

Before force pushing, I'd also rebase on master using `git pull upstream master --rebase`.

**Comment by strodew** [comment]
Right, hopefully that fixes it. Thanks again for the guidance!

File: src/codecoverage/bot/tests/test_artifacts.py [hunk 3]
```diff
@@ -27,6 +27,14 @@ def FAKE_ARTIFACTS_DIR(tmpdir):
     return tmpdir.strpath
 
 
+def test_generate_path(FAKE_ARTIFACTS_DIR):
+    a = ArtifactsHandler([], parent_dir=FAKE_ARTIFACTS_DIR)
+    artifact_jsvm = {'name': 'code-coverage-jsvm.info'}
+    artifact_grcov = {'name': 'code-coverage-grcov.zip'}
```
Comment (marco-c) [inline src/codecoverage/bot/tests/test_artifacts.py]: There are some trailing whitespaces here, which make the build fail. Could you remove them?

Comment (strodew) [reply src/codecoverage/bot/tests/test_artifacts.py]: Yep, booting my VM now. Sorry about that!

Comment (strodew) [reply src/codecoverage/bot/tests/test_artifacts.py]: Had to fetch some changes so it asked me to merge again - hopefully that builds properly now.

**Comment by marco-c** [comment]
Thanks!

**Pull request merged**